api/apiclient_test.go

```diff
@@ -94,7 +95,7 @@ func (s *apiclientSuite) TestConnectWebsocketMultipleError(c *gc.C) {
 	info := s.APIInfo(c)
 	addr := listener.Addr().String()
 	info.Addrs = []string{addr, addr, addr}
-	_, _, err = api.ConnectWebsocket(info, api.DialOpts{})
+	_, _, err = api.DialAPI(info, api.DialOpts{})
 	c.Assert(err, gc.ErrorMatches, `unable to connect to API: websocket.Dial wss://.*/model/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}/api: .*`)
```

Comment:
Again, this is existing code. I'd prefer to avoid mixing too many concerns in this PR.

And although I feel your pain (and I've had the same issue before), I have also had problems when error messages weren't checked properly. I think that, on balance, although it's a pain to change the error messages, it's also good to see in the tests what the actual message will look like (to some degree of accuracy), as that's often the only time you'll get to check that the user will see an appropriate message in an often-rare case.

That is, error messages are important and it's good to see what they actually look like. I've often made changes to error messages because I've asserted them in the tests and they look too awkward.